src/main/java/com/xiaomi/infra/pegasus/client/PConfigUtil.java

```diff
@@ -68,6 +68,7 @@ public static Properties loadConfiguration(String configPath) throws PException
           throw new PException("config resource not found: " + configPath);
         }
         config.load(stream);
+        stream.close();
```

Comment:
if here need manual-close, the above `stream` also need close, maybe you can modify:
```java
      InputStream stream = null;
      if (PConfigUtil.isZkPath(configPath)) {
        stream = new ByteArrayInputStream(PConfigUtil.loadConfigFromZK(configPath));
      } else if (PConfigUtil.isLocalFile(configPath)) {
        stream = new BufferedInputStream(
            new FileInputStream(configPath.substring(PConfigUtil.LOCAL_FILE_PREFIX.length())));
      } else if (PConfigUtil.isResource(configPath)) {
        stream =
            PegasusClient.class.getResourceAsStream(
                configPath.substring(PConfigUtil.RESOURCE_PREFIX.length()));
      } else {
        throw new PException(
            "configPath format error, "
                + "should be local file format as 'file:///path/to/config', "
                + "or zookeeper path format as 'zk://host1:port1,host2:port2,host3:port3/path/to/config', "
                + "or java resource format as 'resource:///path/to/config', "
                + "but actual configPath is "
                + configPath);
      }
      if (stream == null) {
        throw new PException("config resource not found: " + configPath);
      }
      config.load(stream);
      return config;
```

Reply:
Yeh, Got its!